numpy/ma/core.py

```diff
@@ -1296,14 +1296,23 @@ def _recursive_make_descr(datatype, newtype=bool_):
                 # Prepend the title to the name
                 name = (field[-1], name)
             descr.append((name, _recursive_make_descr(field[0], newtype)))
-        return descr
+        newtype = np.dtype(descr)
+
     # Is this some kind of composite a la (np.float,2)
     elif datatype.subdtype:
         mdescr = list(datatype.subdtype)
         mdescr[0] = _recursive_make_descr(datatype.subdtype[0], newtype)
-        return tuple(mdescr)
+        newtype = np.dtype(tuple(mdescr))
+
+    # promote the type to a dtype if otherwise untouched
     else:
-        return newtype
```

Comment:
Every top-level call-site in the file fails to pass in a `dtype` right now (either `MaskType`, `'O'`, or `bool`, I think), so this is absolutely required. Sure, we could change all the callsites, I guess...